R/pkg/R/DataFrame.R

```diff
@@ -3715,3 +3715,34 @@ setMethod("rollup",
             sgd <- callJMethod(x@sdf, "rollup", jcol)
             groupedData(sgd)
           })
+
+#' hint
+#'
+#' Specifies execution plan hint on the current SparkDataFrame.
+#'
+#' @param x a SparkDataFrame.
+#' @param name a name of the hint.
+#' @param ... additional argument(s) passed to the method.
```

Comment:
Scala is even [more cryptic](https://github.com/rxin/spark/blob/b84badc29c5ca315791e4d023bdebb046e0b3b4f/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L1163-L1171) here. I adjusted it a bit, but I think we can revisit this once we have some practical examples.